From 69ac2ca661c27805cadffc01f4cc07af87172164 Mon Sep 17 00:00:00 2001
From: Katsuki Tozawa <katsuki.tozawa.wz@renesas.com>
Date: Mon, 3 Feb 2025 16:30:43 +0900
Subject: [PATCH] add drp property to devicetree

Adapted for SolidRun RZ/V2N kernel fork:
- r9a09g056n44-dev.dts -> rzv2n-sr-som.dtsi
- r9a09g056n44-evk.dts -> r9a09g056n48-rzv2n-evk.dts
---
 arch/arm64/boot/dts/renesas/r9a09g056.dtsi            | 11 +++++++++++
 arch/arm64/boot/dts/renesas/r9a09g056n48-rzv2n-evk.dts |  7 +++++++
 arch/arm64/boot/dts/renesas/rzv2n-sr-som.dtsi          |  7 +++++++
 3 files changed, 25 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r9a09g056.dtsi b/arch/arm64/boot/dts/renesas/r9a09g056.dtsi
index 1234567..abcdefg 100644
--- a/arch/arm64/boot/dts/renesas/r9a09g056.dtsi
+++ b/arch/arm64/boot/dts/renesas/r9a09g056.dtsi
@@ -702,6 +702,17 @@ ostm7: timer@12c03000 {
 			status = "disabled";
 		};

+		drp1: drp1@17000000 {
+			compatible = "renesas,rzv2n-drp";
+			reg =  <0 0x17000000 0 0x1000000>,
+					<0 0x16800000 0 0x400000>;
+			interrupts = <GIC_SPI 914 IRQ_TYPE_LEVEL_HIGH>,
+							<GIC_SPI 915 IRQ_TYPE_LEVEL_HIGH>;
+			//resets = <&cpg R9A09G056_DRPAI_ARESETN>;
+			resets = <&cpg 253>;
+			status = "disabled";
+		};
+
 		cmtw0: timer@11c01800 {
 			compatible = "renesas,rzv2n-cmtw";
 			reg = <0 0x11c01800 0 0x1000>;
diff --git a/arch/arm64/boot/dts/renesas/r9a09g056n48-rzv2n-evk.dts b/arch/arm64/boot/dts/renesas/r9a09g056n48-rzv2n-evk.dts
index 1234567..abcdefg 100644
--- a/arch/arm64/boot/dts/renesas/r9a09g056n48-rzv2n-evk.dts
+++ b/arch/arm64/boot/dts/renesas/r9a09g056n48-rzv2n-evk.dts
@@ -1066,6 +1066,13 @@
 	status = "okay";
 };

+&drp1 {
+    memory-region = <&drp_codec>;
+    memory-oca-region = <&opencva_reserved>;
+    memory-shared-for-drpai-ext-cont = <&shared_drp_reserved>;
+    status = "okay";
+};
+
 &usb20phyrst {
 	status = "okay";
 };
diff --git a/arch/arm64/boot/dts/renesas/rzv2n-sr-som.dtsi b/arch/arm64/boot/dts/renesas/rzv2n-sr-som.dtsi
index 1234567..abcdefg 100644
--- a/arch/arm64/boot/dts/renesas/rzv2n-sr-som.dtsi
+++ b/arch/arm64/boot/dts/renesas/rzv2n-sr-som.dtsi
@@ -403,6 +403,13 @@
 	status = "okay";
 };

+&drp1 {
+    memory-region = <&drp_codec>;
+    memory-oca-region = <&opencva_reserved>;
+    memory-shared-for-drpai-ext-cont = <&shared_drp_reserved>;
+    status = "okay";
+};
+
 &rtc {
     status = "disabled";
 };
--
2.34.1

